serverlessworkflow/modules/ROOT/pages/core/custom-functions-support.adoc

```diff
@@ -456,6 +487,64 @@ You must declare a `functionRef` like the following.
 ----
 <1> The request payload is set in `arguments`.
 
+[WARNING]
+====
+You can't send a CloudEvent if you don't set the `asCloudEvent` attribute to `true`. If you do so, {product_name} will throw an error.
+====
+
+==== Sending a CloudEvent
+
+Given you want to send the following CloudEvent as the payload:
+
+[source,json]
+----
+{
+  "specversion" : "1.0",
+  "type" : "com.github.pull_request.opened",
+  "source" : "https://github.com/cloudevents/spec/pull",
+  "subject" : "123",
+  "id" : "A234-1234-1234",
+  "time" : "2018-04-05T17:31:00Z",
+  "comexampleextension1" : "value",
+  "comexampleothervalue" : 5,
+  "datacontenttype" : "text/xml",
+  "data" : "<much wow=\"xml\"/>"
+}
+----
+
+You must declare a `functionRef` like the following: (Don't forget to set `asCloudEvent` as `true` in function metadata. See <<sending-cloudevents>>.)
+
+[source,json]
+----
+  "states": [
+    {
+      "name": "invokeFunction",
+      "type": "operation",
+      "actions": [
+        {
+          "functionRef": {
+            "refName": "greet",
+            "arguments": { <1>
+                "specversion" : "1.0",
+                "type" : "com.github.pull_request.opened",
+                "source" : "https://github.com/cloudevents/spec/pull",
+                "subject" : "123",
+                "id" : "A234-1234-1234",
```

Comment:
Id can't be auto-generated by the caller (our engine)? It should. It's not a field users are expected to define in runtime.

Reply:
1. If it's auto-generated, users won't be able to define a template. Something like `my_prefix{variable_id}`. Is that what we want?
2. If it's auto-generated, what will we use to generate it? Random, workflow_id, a template with a combination of properties...?

Reply:
It's auto-generated, period.

You can open a JIRA to improve this. If it's not blank, verify if it's an expression. If so, evaluate it. If not, use it. Then you have the template thing if users need it.

But for this feature, as it is today, it should be generated. It's the easier way to be aligned with the spec for now: https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#id

Reply:
Done.